timesketch/lib/analyzers/sessionizer_test.py

```diff
@@ -0,0 +1,236 @@
+            event2 = (ds.get_event('test_index', '1', stored_events=True))
+            self.assertEqual(event2['_source']['session_number'], 1)
+
+    @mock.patch(u'timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_zero_events(self):
+        """Test the behaviour of the sessionizer when given zero events."""
+        with mock.patch.object(SessionizerSketchPlugin,
+                               'event_stream',
+                               return_value=_create_mock_event(0, 0)):
+            index = 'test_index'
+            sketch_id = 1
+            analyser = sessionizer.SessionizerSketchPlugin(index, sketch_id)
+            message = analyser.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of session created: 0')
+
+    @mock.patch(u'timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_one_event(self):
+        """Test the behaviour of the sessionizer when given one event."""
+        with mock.patch.object(SessionizerSketchPlugin,
+                               'event_stream',
+                               return_value=_create_mock_event(0, 1)):
+            index = 'test_index'
+            sketch_id = 1
+            analyser = sessionizer.SessionizerSketchPlugin(index, sketch_id)
+            message = analyser.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of session created: 1')
+            ds = MockDataStore("test", 0)
+            event1 = (ds.get_event('test_index', '0', stored_events=True))
+            self.assertEqual(event1['_source']['session_number'], 1)
+
+
+def _create_mock_event(event_id, quantity, time_diffs=None):
+    """
+    Returns an instance of Event, based on the MockDataStore event_dict
+    example.
+
+    Args:
+        event_id: Desired ID for the Event.
+        quantity: The number of Events to be generated.
+        time_diffs: A list of time differences between the generated Events.
+
+    Returns:
+        A generator of Event objects.
+    """
+
+    if not time_diffs:
+        time_diffs = [0]
+    if quantity < 0:
+        quantity = abs(quantity)
+
+    # If the list of time differences is too small to be compatible with the
+    # quantity of events, then extend the list with the last value for as many
+    # items as necessary.
+    if quantity - len(time_diffs) > 0:
+        time_diffs.extend([time_diffs[len(time_diffs) - 1]] *
+                          (quantity - len(time_diffs)))
+
+    # Setup for Event object initialisation
+    ds = MockDataStore("test", 0)
+    user = User('test_user')
+    sketch = Sketch('test_sketch', 'description', user)
+    label = sketch.Label(label='Test label', user=user)
+    sketch.labels.append(label)
+
+    ts = 1410895419859714
+    event_template = ds.get_event('test', 'test')
+
+    for i in range(quantity):
```

Comment:
I would do some changes to this function to generate a more realistic-looking set of events.

Instead of having `quantity` events separated by `time_diffs`, I would add "batches" of 100 (or any other number) of events, separated by `time_diffs`.

So instead of having (assuming `event_id,timestamp`)

```
event1,0
event2,3000000 # first element of time_diff
event3,4000000 # second element of time_diff
```

I'd have

```
event1,0
event2,1
event3,2
...
event100,3000000 # first element of `time_diff`
event101,3000001
event102,3000002
...
event200,4000000 # second element of time_diff
event201,4000001
```

It shouldn't take much tweaking and it will make the dataset look more realistic, potentially revealing other missed corner-cases.